cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -104,6 +106,13 @@ public static DataprocClient fromConf(DataprocConf conf) throws IOException, Gen
     }
     boolean useInternalIP = false;
 
+    if (conf.isInNetwork()) {
+      useInternalIP = true;
+      Preconditions.checkArgument(!Strings.isNullOrEmpty(network), "The network to which the instance is peered to " +
+        "must be specified when in network option is selected.");
```

Comment:
perhaps instead of 'inNetwork' we should call it 'peeredNetwork' since that is what will be validated? Also, error messages should mention how to correct the problem when possible. Something like:

A network must be specified when the peered network option is set. Please specify a network.